### DISCUSSION TITLE
Editing forward functions of MobileNetV2 or other backboned models

### DISCUSSION THREAD

**Discussion by zeynepozdemir** [open-mmlab/mmclassification#1475] [Q&A] [answered]
I want to use mobilenetv2, one of the backboned models defined in mmpretrain, as a backboned model in mmfewshot.

I explained the problem I had there in the link below.
https://github.com/open-mmlab/mmfewshot/issues/129

1: https://github.com/open-mmlab/mmpretrain/blob/main/mmpretrain/models/backbones/mobilenet_v2.py  

2: https://github.com/open-mmlab/mmfewshot/blob/main/mmfewshot/classification/models/backbones/resnet12.py

In short, the forward part of the code in 1 returns {tuple :1}. It contains {Tensor: (25,1280,3,3).

2 gives direct output as {Tensor:(25,1600)}.

After that, I want to use Ptotypical Networks, so I need to get output as in 2. How can I do it without breaking the model structure?

It is very important for me to solve this critical point, can you help me?

**Comment by Ezra-Yu** [comment]
the `mmpretrain.model.backbone.forward` return a tuple, and `mmfewshot.classification.backbone.forward` just return a tensor.
Try to modify the code of mmfewshot in https://github.com/open-mmlab/mmfewshot/blob/af4fad50b0e00946467244dc8bce4d232d2b79c2/mmfewshot/classification/models/classifiers/base.py#L73

From :
```
        x = self.backbone(img)
```

To:
```
        x = self.backbone(img)[-1]
```

**Comment by zeynepozdemir** [comment, marked as answer]
I got the following error when I applied the technique you said.

![image](https://user-images.githubusercontent.com/6259137/231735244-7b8d657d-30a5-4fed-9d3e-0c87c98b5931.png)

I solved my problem by updating the forward function like this.. Thank You :) 

![image](https://user-images.githubusercontent.com/6259137/231735850-82b5e699-4cbd-4c3f-a63b-8012544915e3.png)
